### DISCUSSION TITLE
Code for humidifier

### DISCUSSION THREAD

**Discussion by Dimbo6022** [RavenSystem/esp-homekit-devices#1905] [Q&A]
good afternoon. please help write the code for the humidifier. it has 4 operating modes - 1, 2, auto, night. I suppose to manage through d1 mini.

**Comment by joddye2** [comment]
This is a code for on/off humidifier.


{
    "c": {
      "o": 1 
      # UART on
    },
    "a": [
      {
        "0": {
            # mode: off
            "r": [
              {
                "g": 15, 
                "v": 0
              }
            ]
          },
          "1": {
            # mode: idle (eg. reached the humidity)
            "r": [
              {
                "g": 15,
                "v": 0
              }
            ]
          },
          "3": {
            # mode: on (humidifying, relay on)
            "r": [
              {
                "g": 15, # relay pin
                "v": 1 # voltage output on
              }
            ]
          },        
        "t": 26, # humidifier without temp sensor
        "b": [
            # input button: toggle (one click on, seconf off)
          {
            "g": 12, # button pin (shorted with gnd)
            "t": 0
          }
        ],
        "g": 2, # humidity sensor pin
        "n": 1, # humidity sensor type (DHT11)
        "w": 1 # humidifier type (1 - humidifier only)
      }
    ]
  }
